packages/frontend-v2/hooks/useStudentWithPlans.ts

```diff
@@ -0,0 +1,34 @@
+import useSWR, {KeyedMutator, SWRResponse} from "swr";
+import { API } from "@graduate/api-client";
+import { GetStudentResponse } from "@graduate/common";
+import { AxiosError } from "axios";
+
+type StudentResponse = Omit<SWRResponse<GetStudentResponse, AxiosError>, "data" | "mutate">;
+
+type UseStudentReturn = StudentResponse & {
+  isLoading: boolean,
+  mutateStudent: KeyedMutator<GetStudentResponse>;
+  student?: GetStudentResponse;
+}
+
+/**
+ * Returns the student with plan using SWR.
+ * @param jwt JWT for authentication to fetch the data from db.
+ * Will later be removed when we switch to cookies.
+ */
+export function useStudentWithPlans(jwt: string): UseStudentReturn {
+  const key = `api/students/me`;
```

Comment:
idk it's any priority for now since it's just for testing. I think later on it might be better to make a util function to construct the key for hooks. It's just easier to keep track of.